feature_tests/js/test/result-ts.mts

```diff
@@ -0,0 +1,24 @@
+import test from 'ava';
+import { FFIError } from '../api/diplomat-runtime.js';
+import { ErrorEnum } from '../api/ErrorEnum.js';
+import { ErrorStruct } from '../api/ErrorStruct.js';
+
+import { ResultOpaque } from "../api/ResultOpaque.js"
+
+test("Verify result methods", t => {
+    const s = ResultOpaque.new(5);
+    s.assert_integer(5);
+
+    const error_foo = t.throws(() => ResultOpaque.new_failing_foo()) as unknown as FFIError<ErrorEnum>;
```

Comment:
Nit: We should attempt to avoid `as unknown` with our clean TypeScript types

Reply:
This is TypeScript being bad at exceptions, since it's not possible to catch an error of a specific type without a ton of extra work. This makes ava have to return some basic `Error` type, but even if we did have our own `try { .. } catch (e) { .. }` stuff we still couldn't type check.